geoh5py/objects/object_base.py

```diff
@@ -136,26 +163,25 @@ def add_data(
                 )
 
             attr["name"] = name
+
             attr, validate_property_group = self.validate_association(
                 attr, property_group=property_group, **kwargs
             )
 
             entity_type = self.workspace.validate_data_type(attr, attr.get("values"))
 
-            kwargs = {"parent": self, "association": attr["association"]}
+            kwargs = {"parent": self}
             for key, val in attr.items():
-                if key in ["parent", "association", "entity_type", "type"]:
-                    continue
-
-                kwargs[key] = val
+                if key not in ["parent", "entity_type", "type"]:
+                    kwargs[key] = val
 
             data_object = self.workspace.create_entity(
-                Data, entity=kwargs, entity_type=entity_type, compression=compression
+                Data,  # type: ignore
+                entity=kwargs,
+                entity_type=entity_type,
+                compression=compression,
             )
 
-            if not isinstance(data_object, Data):
-                continue
-
```

Comment:
Same

Reply:
when passing "Data" to create_entity, it's redudndant to check if the outptut is "Data" no?